src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -95,16 +97,19 @@
     private final Map<MqttTopic, AwsIotMqttClient> subscriptionTopics = new ConcurrentHashMap<>();
     private final AtomicInteger connectionRoundRobin = new AtomicInteger(0);
     private final AtomicBoolean mqttOnline = new AtomicBoolean(false);
+    CompletableFuture<Boolean> waitMqttOnline = new CompletableFuture<>();
 
     private final EventLoopGroup eventLoopGroup;
     private final HostResolver hostResolver;
     private final ClientBootstrap clientBootstrap;
     private final CallbackEventManager callbackEventManager = new CallbackEventManager();
     private final Spool spool;
-    private final ScheduledExecutorService ses;
+    protected final ScheduledExecutorService ses;
```

Comment:
you aren't using the ses.

Reply:
this is used to reconnect the connection, I am considering whether to change the executor to the executorService as well?

Reply:
That's fine to keep then, but why did you change the visibility of it.

Reply:
the format check would complain if I keep it as private

Reply:
What check? It was private before.